### PR TITLE
Exclude Beams from backing up any content

### PR DESCRIPTION
**Pull request by daniellevass**
This should fix issue #86 and prevent the library from creating any backups at all.

### CODE REVIEW

File: pushnotifications/src/main/res/xml/backup_rules.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<full-backup-content>
+    <exclude domain="sharedpref" path="."/>
```
Comment (luismfonseca) [inline pushnotifications/src/main/res/xml/backup_rules.xml]: Does this include that file where we resume the operations?

**Comment by daniellevass** [comment]
Have updated this to exclude only the push notifications shared preferences. There's an open question for how we want to handle our jobqueue.

You can test this works for yourself by entering `adb shell` then running `bmgr run` and then `bmgr fullbackup com.pusher.pushnotifications.sample`. Check in the Android Logcat to see what files got backed up. 

Using this we can now demonstrate that the shared preferences file is no longer included!

**Comment by daniellevass** [comment]
Added some more logic to move our jobqueue to a directory and exclude that.

Files backed up before:

```xml
Name: apps/com.pusher.pushnotifications.sample/f/beams
Name: apps/com.pusher.pushnotifications.sample/f/beams/8a070eaa-033f-46d6-bb90-f4c15acc47e1.jobqueue
Name: apps/com.pusher.pushnotifications.sample/db/google_app_measurement_local.db-journal
Name: apps/com.pusher.pushnotifications.sample/db/google_app_measurement_local.db
Name: apps/com.pusher.pushnotifications.sample/sp/com.google.android.gms.appid.xml
Name: apps/com.pusher.pushnotifications.sample/sp/com.google.android.gms.measurement.prefs.xml
```

Files backed up after:

```xml
Name: apps/com.pusher.pushnotifications.sample/db/google_app_measurement_local.db-journal
Name: apps/com.pusher.pushnotifications.sample/db/google_app_measurement_local.db
Name: apps/com.pusher.pushnotifications.sample/sp/com.google.android.gms.appid.xml
Name: apps/com.pusher.pushnotifications.sample/sp/com.google.android.gms.measurement.prefs.xml
```

File: pushnotifications/src/main/java/com/pusher/pushnotifications/internal/ServerSyncHandler.kt [hunk 2]
```diff
@@ -76,7 +76,9 @@ class ServerSyncHandler private constructor(
           val moshi = Moshi.Builder().add(ServerSyncJob.jsonAdapterFactory).build()
           val converter = MoshiConverter(moshi.adapter(ServerSyncJob::class.java))
 
-          val jobQueue = TapeJobQueue<ServerSyncJob>(File(secureFileDir, "$instanceId.jobqueue"), converter )
+          File(secureFileDir, "beams").mkdirs()
+          val file = File(secureFileDir, "beams/$instanceId.jobqueue")
+          val jobQueue = TapeJobQueue<ServerSyncJob>(file, converter )
```
Comment (luismfonseca) [inline pushnotifications/src/main/java/com/pusher/pushnotifications/internal/ServerSyncHandler.kt]:
```suggestion
          val jobQueue = TapeJobQueue<ServerSyncJob>(file, converter)
```

File: pushnotifications/src/main/res/xml/backup_rules.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<full-backup-content>
+    <exclude domain="sharedpref" path="com.pusher.pushnotifications.PushNotificationsInstance.xml" />
+    <exclude domain="file" path="beams/." />
+</full-backup-content>
```
Comment (luismfonseca) [inline pushnotifications/src/main/res/xml/backup_rules.xml]:
```suggestion
</full-backup-content>

```

**Pull request merged**